Add definition of nRF52805 SoC and nrf52dk_nrf52805 board

anangl opened the pull request:
Add definition of the nRF52805 SoC together with the corresponding dts files.

anangl committed:
soc: nordic_nrf: Add definition of nRF52805 SoC

Add definition of the nRF52805 SoC together with the corresponding
dts files.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

ioannisg reviewed (commented):
Looks ok to me, didn't spot any issues, however, I would like to see a board definition, if possible

anangl committed:
boards: arm: Add definition of nrf52dk_nrf52805 board

Add a board that uses the nRF52805 SoC.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

ioannisg commented on boards/arm/nrf52dk_nrf52805/CMakeLists.txt:
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# The nrf52dk_nrf52805 board mirrors the nRF52832 DK hardware. This
```
Strictly speaking, there is no "nRF52832 DK" hardware. I know we have the same line in nrf52810, but being pedantic, this should be nRF 52 DK

anangl replied:
Fixed.

ioannisg reviewed (approved):
Looks fine to me

ioannisg commented on dts/arm/nordic/nrf52805.dtsi:
```diff
@@ -0,0 +1,223 @@
+		};
+
+/* Disabled until MDK is updated to provide QDEC definition and bitfields. */
+#if 0
```
Why was this needed? Did it result in an build errors?

anangl replied:
Yes, the build assertion checking the base address failed here:
https://github.com/zephyrproject-rtos/zephyr/blob/b046ca54093abbede73d507afa661a8470dd352a/soc/arm/nordic_nrf/validate_base_addresses.c#L104

ioannisg reviewed (commented):
@anangl the nrf52805 should be "CAAA" not QFAA

anangl commented:
> @anangl the nrf52805 should be "CAAA" not QFAA

Fixed.

ioannisg reviewed (changes requested):
One more question @anangl , you do not seem to introduce an LE Radio header for this SoC. I wonder how this actually works, do the LE samples build?

cvinayak commented:
> 
> 
> One more question @anangl , you do not seem to introduce an LE Radio header for this SoC. I wonder how this actually works, do the LE samples build?

If the BLE Radio headers have same values as that of nRF52832, then only a conditional include of existing nRF52832 header should suffice. nRF52 Series is starting the get variants with same Radio peripheral/characteristics, duplicate header files can be avoided.

carlescufi commented:
Please update the SHA in west.yml

anangl commented:
> Please update the SHA in west.yml

Done.

mbolivar-nordic commented:
> #### Conflicting files
> `west.yml`

needs rebase

anangl commented:
Rebased on master.

anangl commented:
DNM until BLE support is added.

anangl committed:
bluetooth: controller: nrf5: Refactor a bit radio_nrf5_ppi.h

Change hard-coded (D)PPI channel bitfield values to the BIT() macro
calls with the corresponding channel base value supplied as parameter,
to make it possible to actually change this base value if needed.
Remove unused *_EXCLUDE macros.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
bluetooth: controller: nrf5: Add support for nRF52805

Add the nRF52805-specific file with radio related defintions.
Use also specific configurations for certain PPI channels used by
the controller, as this SoC has limited number of programmable PPI
channels.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl commented:
@cvinayak Could you take a look at the Bluetooth related stuff (last two commits)?

cvinayak reviewed (approved):
Looks good to me. Have tested the samples/bluetooth/peripheral built for nrf52dk_nrf52805 on nrf52dk_nrf52832 with increased RAM manually in Kconfig. No issues noticed with the changes to use PPI 2, 3 and 4 for sw tIFS switch (instead of normally used 10, 11 and 12 on nRF52832).

carlescufi commented:
@ioannisg please re-review

ioannisg commented on soc/arm/nordic_nrf/nrf52/Kconfig.defconfig.nrf52805_QFAA:
```diff
@@ -0,0 +1,14 @@
+# Nordic Semiconductor nRF52805 MCU
```
@anangl shouldn't this file be named with the right suffix, CAAA?

anangl replied:
Of course it should. Thanks for catching this!

anangl replied:
Fixed.

ioannisg commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_ppi.h:
```diff
@@ -369,7 +369,15 @@ static inline void hal_sw_switch_timer_clear_ppi_config(void)
+#if (EVENT_TIMER_ID == 0)
+#define HAL_SW_SWITCH_GROUP_TASK_ENABLE_PPI 2
+#else
+#error "Insufficient resources"
```
I guess you should explain the reason for the "insufficient resources", either in a comment, or directly in the preprocessor #error 

Hard to see how the insufficient resources relates to the ID of the Event TImer.

anangl replied:
I'll add some more descriptive message/comment.

anangl replied:
Added.

ioannisg commented:
> @ioannisg please re-review

Just did that @carlescufi 
I've just got 2 comments/questions for @anangl then, I guess, it's good to go.

ioannisg reviewed (approved):
thanks!
